src/internal/functions/Add-DscResourceModule.ps1

```diff
@@ -0,0 +1,58 @@
+function Add-DscResourceModule {
+  <#
+  .SYNOPSIS
+    Vendor PowerShell modules into the build folder either by path or by retrieving the modules from the Gallery
+
+  .DESCRIPTION
+    Given a DSC Resource Info object, introspect on the source (if possible) to return information about
+    the resources parameters, including their help text, whether or not they are mandatory for calling the
+    resource with the Get method, and whether or not they are mandatory for calling the resource with the
+    Set method.
+
+    This information can be reliably retrieved if the DSC resource is implemented in PowerShell and is
+    not class based. Otherewise, best guesses are made about the mandatory status of the parameters via
+    the Required attribute and the help information is null.
+
+  .PARAMETER Name
+    Specifies the name of the module to save.
+
+  .PARAMETER Path
+    Specifies the location on the local computer to store the saved module.
+
+  .PARAMETER RequiredVersion
+    Specifies the exact version number of the module to save.
+    If left blank, will default to latest available.
+
+  .EXAMPLE
+    Add-DscResourceModule -TargetDir ./tmp -Name PowerShellGet -Version 2.2.3
+
+  #>
+  [CmdletBinding()]
+  param (
+    $Name,
+    $Path = './tmp',
+    $RequiredVersion
+  )
+
+  Begin { }
+
+  Process {
+    $PathTmp = "$($Path)_tmp"
+
+    if (-not(Test-Path $Path)) {
+      if (-not(Test-Path $Path)) {
+        mkdir $Path
```

Comment:
Though the original script used `mkdir` I think we'll want to use `New-Item` to avoid any aliases in our code:

```suggestion
        New-Item -Path $Path -Force -Type Directory
```